lib/watir/radio_set.rb

```diff
@@ -0,0 +1,218 @@
+module Watir
+  class RadioSet
+    extend Forwardable
+    include Watir::Exception
+    include Enumerable
+
+    delegate [:exists?, :present?, :visible?, :browser] => :source
+
+    attr_reader :source, :frame
+
+    def initialize(query_scope, selector)
+      unless selector.kind_of? Hash
+        raise ArgumentError, "invalid argument: #{selector.inspect}"
+      end
+
+      @source = Radio.new(query_scope, selector)
+      @frame = @source.parent(tag_name: :form)
+    end
+
+    #
+    # Yields each TableCell associated with this row.
+    #
+    # @example
+    #   radio_set = browser.radio_set
+    #   radio_set.each do |radio|
+    #     puts radio.text
+    #   end
+    #
+    # @yieldparam [Watir::RadioSet] element iterate through the radio buttons.
+    #
+
+    def each(&block)
+      radios.each(&block)
+    end
+
+    #
+    # Get the n'th radio button in this set
+    #
+    # @return Watir::Radio
+    #
+
+    def [](idx)
+      radios[idx]
+    end
+
+    #
+    # @return Watir::Radio
+    #
+
+    def radio(opt)
+      name_opt = name.empty? ? {} : {name: name}
+      frame.radio(opt.merge name_opt)
+    end
+
+    #
+    # @return Watir::RadioCollection
+    #
+
+    def radios
+      opt = name ? {name: name} : {}
+      element_call(:wait_for_present) { frame.radios(opt) }
+    end
+
+    #
+    # Returns true if any radio buttons in the set are enabled.
+    #
+    # @return [Boolean]
+    #
+
+    def enabled?
+      any?(&:enabled?)
+    end
+
+    #
+    # Returns true if all radio buttons in the set are disabled.
+    #
+    # @return [Boolean]
+    #
+
+    def disabled?
+      !enabled?
+    end
+
+    #
+    # Returns true if all radio buttons in the set are disabled.
+    #
+    # @return [String]
+    #
+
+    def name
+      @name ||= source.name
+    end
+
+    #
+    # If RadioSet exists, this always returns 'radio'.
+    #
+    # @return [String]
+    #
+
+    def type
+      source.send :assert_exists
+      'radio'
+    end
+
+    #
+    # Returns true if the radio set has one or more radio buttons where label matches the given value.
+    #
+    # @param [String, Regexp] str_or_rx
+    # @return [Boolean]
+    #
+
+    def include?(str_or_rx)
+      radio(label: str_or_rx).exist?
+    end
+
+    #
+    # Select the radio button whose value or label matches the given string.
+    #
+    # @param [String, Regexp] str_or_rx
+    # @raise [Watir::Exception::UnknownObjectException] if the Radio does not exist.
+    # @return [String] The value or text of the radio selected.
+    #
+
+    def select(str_or_rx)
```

Comment:
Oh? Watir Classic would match select options by either text or value? Yes, I'd like to see it changed in Select. Page Object gem actually recently implemented something to support that for select lists in their form filling method.